src/v/model/metadata.h

```diff
@@ -624,6 +626,45 @@ enum class recovery_validation_mode : std::uint16_t {
 std::ostream& operator<<(std::ostream&, recovery_validation_mode);
 std::istream& operator>>(std::istream&, recovery_validation_mode&);
 
+enum class license_required_feature : uint8_t {
```

Comment:
any reason to keep this in model vs features? also, if it stays here, consider using a separate header. we should start making things more fine grained now that bazel is around.

Reply:
not particularly. I occasionally struggle to find the "correct" place for things, so if your intuition suggests `features` I'm happy to move it there.

Reply:
i might move it, if for any other reason than `model` isn't very well defined and we keep putting stuff in it that doesn't have a good home. features seem to at least be semi related. but either way its not a big deal.

Reply:
done